fairscale/optim/oss.py

```diff
@@ -146,9 +146,22 @@ def state_dict(self) -> Dict[str, Any]:
             len(self._all_states) > 0
         ), "The optimizer state is not materialized, please call consolidate_state_dict on every replica beforehand"
 
+        # Flatten the state_dict, save the partition
+        # The partition indexes the flat param_groups list,
+        # and logs the rank <> shard correspondence
+        partition: Dict[int, Tuple[int, int]] = {}
+        param_groups: List[Dict[Any, Any]] = []
+
+        i_global = 0
```

Comment:
This name is confusing. Maybe call it `start` instead?

Reply:
ah, it was meant as "global index", but start works I guess